### DISCUSSION TITLE
Refresh token is not working properly

### DISCUSSION THREAD

**Discussion by patriarcholeksiy** [LeleDerGrasshalmi/FortniteEndpointsDocumentation#231] [Q&A] [answered]
When I receive the token using the authorization code, the response indicates that the upgrade token is available for renewal within 8 hours, but when I try to use it, an error appears: 400 Invalid request: "{"Error code":"errors.com.epicgames.account.auth_token.invalid_refresh_token","ErrorMessage":"Sorry update token "XXXXXXXXXXXXXXX" is invalid","messageVars":["XXXXXXXXXXXXX"],"numericErrorCode":18036,"originatingService":"com.epicgames.account.public","intent":"prod","error_description":"Sorry, the update token "XXXXXXXXXXXXXXX" is invalid","error":"invalid_grant"}"

If I use refresh_token for 30 minutes, then it works fine. Can you tell me what to do? Thank you in advance!

**Comment by LeleDerGrasshalmi** [comment, marked as answer]
refresh tokens work properly, its just that when you launch fortnite all auth sessions will be killed (so also refresh tokens)

**Reply by patriarcholeksiy** [reply]
Is there a way around this?

**Reply by LeleDerGrasshalmi** [reply]
Thats how authentication works, no.
